lib/commands/appmanager-livesync.ts

```diff
@@ -0,0 +1,42 @@
+﻿///<reference path="../.d.ts"/>
+"use strict";
+
+export class AppManagerLiveSyncCommand implements ICommand {
+	constructor(private $prompter: IPrompter,
+		private $mobileHelper: Mobile.IMobileHelper,
+		private $appManagerService: IAppManagerService,
+		private $errors: IErrors) { }
+
+	public execute(args: string[]): IFuture<void> {
+		return ((): void => {
+			if(!args || args.length === 0) {
+				var shouldLivePatchAllPlatforms = this.$prompter.confirm("This command will publish livepatch version to all appstores. Are you sure you want to continue?",() => true).wait();
+				if(shouldLivePatchAllPlatforms) {
+					this.$appManagerService.publishLivePatchForAllPlatforms().wait();
+				}
+			} else {
+				this.$appManagerService.publishLivePatch(args).wait();
+			}
+		}).future<void>()();
+	}
+
+	public canExecute(args: string[]): IFuture<boolean> {
+		return ((): boolean => {
+			var lowerCasedArguments = _.map(args, arg => arg.toLowerCase());
+
+			_.each(args, platform => this.$mobileHelper.validatePlatformName(platform));
+
+			// make sure each platform is specified only once
+			_.each(lowerCasedArguments, platform => {
```

Comment:
I'm also considering `_.countBy` but I'm also not sure which of the three (current implementation, `_.countBy`, `_.groupBy`) is the best solution.